Frontend release august vol 2 (#440)
SenseNet/sensenet.github.io: _updates/2020-08-26-august.md

---
title: August 2020 / 2
---

## Bye bye summer, hello autumn!🍃🍂

> We have to say goodbye to summer, but something new has begun: autumn. The ["ember months"](https://en.wiktionary.org/wiki/ember_months) hold new challenges, suprises and of course new tasks for us. We face them with passion in the future as we have done so far. We are really excited of Hacktoberfest, we already started working on the contributor's tasks. In the upcoming months we will focus on bug fixin' instead of implementing new features.

Reviewer: ```suggestion
We have to say goodbye to summer, but it is also a new beginning of something: autumn. The ["ember months"](https://en.wiktionary.org/wiki/ember_months) hold new challenges, surprises and of course new tasks for us. We deal them with passion in the future as we have done so far. We are really excited about Hacktoberfest, so we already started to work on tasks for contributing. In the upcoming months we will focus on bug fixin' instead of implementing new features.
```

> Here are the changes what we done:

Reviewer: ```suggestion
Here are the changes what we've done:
```

### Admin-ui@1.11.0 

- We had to downgrade our date-io dependency version to 1.3.13 cause with newer versions we had some issues.😢
 [#788](https://github.com/SenseNet/sn-client/pull/788)

> - We did some changes on avatar as well, now users can upload images and set it as their avatars with a usage of upload dialog. This process is more simple than before. Refreshing is not working properly now, but keep calm: we are working on it. 😉

Reviewer: ```suggestion
- We did some changes on avatar as well, now users can upload images and set it as their profile pictures using the upload dialog. The process is easier than before. Refreshing is not working properly for now, but keep calm: we are working on it. 😉
```

 [#740](https://github.com/SenseNet/sn-client/pull/740)

![AvatarUpload](/img/avatar_upload_with_dialog.gif "AvatarUpload")

> - Our biggest new feature (in the last 2-3 weeks) is the CTD creator 🎉🎆

Reviewer: ```suggestion
- The biggest new feature of the last 2-3 weeks is binary based content creation 🎉🎆
```

Adding a new CTD, Settings or Localization file was a nightmare before, but you can easily do it now. You will have a custom editor (a monaco texteditor) with a default template that you can use as a helper sheet to create your new content type. What you have to do now is just click on the + (Add) button on the 'Content Types' page, select Content type from the dropdown list and fill the .xml file with your expectation. And voila... 🎺⛲🎯
You can also create new Localization and Settings file like this.
 [#784](https://github.com/SenseNet/sn-client/pull/784)

![CreateCTD](/img/create_ctd.gif "CreateCTD")

> - We have a brand new dashboard.🚅 You can check it right after login, cause this is our landing page as well. You can now see the actual usage of your repository and following the links on the bottom you can learn more about sensenet. 🌎🌌 On our header we also make the repository url visible to know exactly which repo are you in. 

Reviewer: ```suggestion
- The admin surface has a brand new dashboard.🚅 You can check it right after login, cause this is the landing page as well. You can now see the actual usage and limitations of your repository and following the links on the bottom you can learn more about sensenet. 🌎🌌 In the header we also made the repository url visible to know exactly which repository are you in. 
```

![Dashboard](/img/dashboard.png "Dashboard")
![Dashboard2](/img/dashboard2.png "Dashboard2")

- Last but not least we have implemented the restore function in trash. If you accidentally delete something important, you can restore it with 2 mouse click.
In addition you can change the location of the restored file if you would like to move it to different folder from its original place. 💾⏪
[#792](https://github.com/SenseNet/sn-client/pull/792)

![Restore](/img/restore_from_trash.gif "Restore")

### @sensenet/controls-react@4.0.0

- We performed a large refactoring on the view controls and their display interfaces, which also affected our exapmle apps and our sn-controls-react package.
 [#764](https://github.com/SenseNet/sn-client/pull/764)

### @sensenet/document-viewer-react@2.3.0

> - We had a bug related to the preview. Don't worry! All the bugs flew away 🐛🐞

Reviewer: ```suggestion
- There was a bug related to the preview generation. Don't worry! All the bugs flew away 🐛🐞
```

[#791](https://github.com/SenseNet/sn-client/pull/791)

### @sensenet/client-core@3.3.0

> - RegeneratePreviews action has now a return value wich contains PageCount and PreviewCount numbers.

Reviewer: ```suggestion
- RegeneratePreviews action has now a return value which contains `PageCount` and `PreviewCount` numbers.
```
